### DISCUSSION TITLE
why doesnt webui use a VENV like automatic?

### DISCUSSION THREAD

**Discussion by gsgoldma** [oobabooga/text-generation-webui#462] [General]
wouldnt it keep all the packages contained in the root folder that way?

**Comment by BetaDoggo** [comment]
Conda uses less storage and probably has some other benefits.

You can manually install it into a venv if you want to but it doesn't really have any advantages compared to the autoinstallers which already create the conda environment inside the folder.

**Comment by Alcyon6** [comment]
I just use my SD venv to run Oobabooga. Is anyone else doing this? Figure it could save having a bunch more copies of dependencies installed as duplicates. Haven't managed to get the 4-bit 30B model to run yet but I don't think that's why. The smaller ones work fine.

**Reply by Wolvensdale** [reply]
Since the dependencies of each repo is different isn't it dangerous? Might have a case when one repos need something that will make other repos won't work. Will there be any case of that? Or it should never conflict?

**Reply by slopcop** [reply]
Yea, I've had issues with Tortoise screwing up Auto1111 which is apparently screwing this up too :|

**Comment by TrincaMirko** [comment]
i installed it using that exact principle on my windows machine without any conda or wsl like that. It keeps all clean and neat if i need to purge it or move it for some reason

**Reply by ke1ne** [reply]
> i installed it using that exact principle on my windows machine without any conda or wsl like that. It keeps all clean and neat if i need to purge it or move it for some reason

It would be great if you find a couple of minutes and describe the process to other users. At least, something like a command line history. Cheers

**Reply by BetaDoggo** [reply]
Make sure you have git and python installed on path.
On windows you would run the following in powershell:
``` 
git clone https://github.com/oobabooga/text-generation-webui.git
cd text-generation-webui
python -m venv venv
.\venv\Scripts\activate
<torch install command for your hardware from https://pytorch.org/get-started/locally/>
pip install -r .\requirements.txt
```
Afterwards just run .\venv\Scripts\activate before running your webui startup command.
If you're on nvidia you will also need to do the windows 8bit fix detailed on the [wiki](https://github.com/oobabooga/text-generation-webui/wiki/Windows-installation-guide).

On Linux it's the same except for the enter venv command which is 
`source .\venv\bin\activate`

**Reply by ke1ne** [reply]
Thank you!

**Comment by slopcop** [comment]
I've been using a conda 'textgen' environment, and it seems to be working fine.  Between Auto1111, Tortoise, this and Vits, if I mix anything up, the rest break. It's similar to whack a mole. I learned this the hard way, and I now have conda envs for each. Anyways..
 Is the naming, "venv", specifically important? I've been stuck on a 'you need tensorboard' error for days, so maybe this is the cause.  I've tried everything I can think of.

**Reply by jllllll** [reply]
> I've been using a conda 'textgen' environment, and it seems to be working fine. Between Auto1111, Tortoise, this and Vits, if I mix anything up, the rest break. It's similar to whack a mole. I learned this the hard way, and I now have conda envs for each. Anyways.. Is the naming, "venv", specifically important? I've been stuck on a 'you need tensorboard' error for days, so maybe this is the cause. I've tried everything I can think of.

The closest thing I've made to just having everything under a single environment folder is this: https://github.com/jllllll/one-click-installers/tree/oobabooga-windows-conda-bootstrap

It uses Micromamba to create a conda environment, installs conda into it without soft/hard-links, and then Micromamba gets deleted. This leaves a fully independent conda environment contained in a single folder.

Obviously, that script is for Windows, but it shouldn't be too hard to make a bash version.

**Comment by St33lMouse** [comment]
Why would you share a venv or a conda between these different installations? It violates the whole concept of keeping environments separate so all their convoluted dependencies can work. Are you that hard up for space?

You may get away with it, or you may give yourself a major headache.

One thing you can do is share models across installations. Just make a symlink of the model folder so you don't to make multiple copies of these enormous model files.
